src/create-glamorous.js

```diff
@@ -37,40 +73,46 @@ function createGlamorous(splitProps) {
      *   props and the return value is used.
      * @return {ReactComponent} the ReactComponent function
      */
-    function glamorousComponentFactory(...styles) {
+    function glamorousComponentFactory(...styles: Array<Style>) {
       /**
        * This is a component which will render the comp (closure)
        * with the glamorous styles (closure). Forwards any valid
        * props to the underlying component.
        */
-      const GlamorousComponent = withTheme(
+      let FinalGlamorousComponent // eslint-disable-line prefer-const
```

Comment:
It actually is reassigned, on line 152. I'm not sure why eslint wasn't detecting this properly.

We need a binding that will eventually contain the class because the component we pass into `withTheme` needs access to it. So this part is a little bit awkward.